### ISSUE TITLE
np.arange seems to be unsupported with exact dtype argument

### ISSUE THREAD

**Issue by kozlov-alexey** [numba/numba#4659]
Hi,

I noticed that the following simple test fails to run if Numba is used in nopython mode.
The issue seems to be related to a call to np.arange with dtype specified as an argument. 

I was using Numba: 
numba                     0.46.0dev0      np116py37h6538335_82    numba/label/dev

The test itself:
```
    def test_numba_arange_with_dtype(self):
        def test_impl(n):
            arr = np.arange(n, dtype=np.int64)
            return arr
        jitted_func = numba.njit(test_impl)
        n = 11
        np.testing.assert_array_equal(jitted_func(n), test_impl(n))

```
The exact error is:
> Traceback (most recent call last):
>   File "C:\Users\akozlov\AppData\Local\Continuum\anaconda3\hpat\hpat\tests\test_serie
> s.py", line 1710, in test_numba_arange_with_dtype
>     np.testing.assert_array_equal(jitted_func(n), test_impl(n))
>   File "C:\Users\akozlov\AppData\Local\Continuum\anaconda3\envs\HPAT\lib\site-package
> s\numba\dispatcher.py", line 376, in _compile_for_args
>     error_rewrite(e, 'typing')
>   File "C:\Users\akozlov\AppData\Local\Continuum\anaconda3\envs\HPAT\lib\site-package
> s\numba\dispatcher.py", line 343, in error_rewrite
>     reraise(type(e), e, None)
>   File "C:\Users\akozlov\AppData\Local\Continuum\anaconda3\envs\HPAT\lib\site-package
> s\numba\six.py", line 658, in reraise
>     raise value.with_traceback(tb)
> numba.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
> Invalid use of Function(<built-in function arange>) with argument(s) of type(s): (int
> 64, dtype=class(int64))
>  * parameterized
> In definition 0:
>     AssertionError:
>     raised from C:\Users\akozlov\AppData\Local\Continuum\anaconda3\envs\HPAT\lib\site
> -packages\numba\typing\npydecl.py:632
> In definition 1:
>     AssertionError:
>     raised from C:\Users\akozlov\AppData\Local\Continuum\anaconda3\envs\HPAT\lib\site
> -packages\numba\typing\npydecl.py:632
> This error is usually caused by passing an argument of a type that is unsupported by
> the named function.
> [1] During: resolving callee type: Function(<built-in function arange>)
> [2] During: typing of call at C:\Users\akozlov\AppData\Local\Continuum\anaconda3\hpat
> \hpat\tests\test_series.py (1706)
> 

Can you please fix that?

Best Regards,
Alexey.

**Comment by stuartarchibald** [comment]
Thanks for the report. What you are doing (dtype kwarg in `np.arange`) is not supported, so this is a feature request. I just took a look at the `np.arange` code in Numba, it's using the old APIs and needs an update. FWIW, making the call in this specific manner will work:
```
In [10]: @njit 
    ...: def foo(): 
    ...:     return np.arange(1, 10, 1, np.uint8) 
```
that is, all of `start, stop, step` supplied, and the dtype supplied as an arg not a kwarg.

**Comment by guilhermeleobas** [comment]
Hi @stuartarchibald, the old API is `@lower_builtin`? And the new one would be `@overload`?

**Comment by stuartarchibald** [comment]
@guilhermeleobas yes, well, "old" may be a little excessive, it's more that the `numba.extending.*` API is preferred.

**Comment by stuartarchibald** [comment]
Closing, fixed in #4770
